File: Day 01/task.py
list1, list2 = [], []
for line in open("input.txt","r").read().splitlines():
  list1.append(int(line.split("   ")[0]))
  list2.append(int(line.split("   ")[1]))
list1.sort()
list2.sort()
count, count2 = 0, 0
for i in range(len(list1)):
  occurances = 0
  count += abs(list1[i] - list2[i])
  for j in range(len(list2)):
    if list1[i] == list2[j]:
      occurances += 1
  count2 += list1[i] * occurances
print(count, " ", count2)

File: Day 03/task.pl
#!/usr/bin/perl
use strict;
use warnings;

my $path = "input.txt";
open my $file, '<', $path or die "Could not open $path: $!";

my $fullText = "";
while (my $line = <$file>) {
    chomp $line;
    $fullText .= $line;
}

$fullText =~ s/(don't\(\))(.*?)(do\(\))//g; #Remove this line for Part 1 Solution
my @matches = $fullText =~ /mul\(([0-9]{1,3},[0-9]{1,3})\)/g;

my $count = 0;
while(my $element = shift(@matches)) {
    my @results = split(",", $element);
    $count += $results[0] * $results[1];
}

print "The result is: $count\n";
